compiler/rustc_resolve/src/late.rs

```diff
@@ -2902,7 +2902,16 @@ impl<'a, 'ast, 'ra, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
                     }
                 }
 
-                if param.ident.name == kw::UnderscoreLifetime {
+                // To avoid emitting two similar errors, we need to check if the span is a raw underscore lifetime, see issue #143152
+                let is_raw_underscore = self
+                    .r
+                    .tcx
+                    .sess
+                    .psess
+                    .raw_identifier_spans
+                    .iter()
+                    .any(|span| span == param.span());
+                if param.ident.name == kw::UnderscoreLifetime && !is_raw_underscore {
```

Comment:
You should still be able to `emit_unless(is_raw_underscore)`. I don't think calling `self.record_lifetime_param(param.id, LifetimeRes::Error);` matters if it's a raw underscore or not.

Reply:
i.e.

```
if param.ident.name == kw::UnderscoreLifetime {
  let is_raw_underscore = self
    .r
    .tcx
    .sess
    .psess
    .raw_identifier_spans
    .iter()
    .any(|span| span == param.span());
    self.r
      .dcx()
      .create_err(errors::UnderscoreLifetimeIsReserved { span: param.ident.span })
      .emit_unless(is_raw_lifetime);
  // Record lifetime res, so lowering knows there is something fishy.
  self.record_lifetime_param(param.id, LifetimeRes::Error);
  continue;
}
```